### ISSUE TITLE
Sending numbers instead of string in sendEvent segments causes break on Android

### ISSUE THREAD

**Issue by florianf22** [Countly/countly-sdk-react-native-bridge#201]
Best described in the example below:

```
const event = { eventName: 'SMA Type', eventCount: 1, segments: { SMAType: smaType } };
Countly.sendEvent(event);
```

SmaType's an example, if smaType in segments was a number, not a string, it would fail.
![image](https://github.com/Countly/countly-sdk-react-native-bridge/assets/72348409/f200766c-e98e-452f-b0fa-e6000eceeef2)

Is this desired behavior?

**Comment by peterbrown0** [comment]
Hello, thank you for reporting this. Which SDK version are you using?

**Comment by florianf22** [comment]
> Hello, thank you for reporting this. Which SDK version are you using?
20.11.0

**Comment by ArtursKadikis** [comment]
Hello, after looking into it, it looks like there is a bug that prevents handling numbers as segmentation values.

Thank you for bringing this to our attention. For now there are no workarounds other then providing the values as string.

We will work on a fix for this and release it within the next few weeks.

**Comment by peterBrxwn** [comment]
Hello @florianf22, this has been fixed and released in version 24.4.0
Kindly remember to use the new event interface `Countly.events.recordEvent(...)`

Feel free to reopen this issue if you are still experiencing this.
